src/allreduce_robust.h

```diff
@@ -286,12 +284,9 @@ class AllreduceRobust : public AllreduceBase {
         int role_flag = src[i].flag(SeqType::kCache) & dst[i].flag(SeqType::kCache);
         // if seqno is different in src and destination
         int seq_diff_flag = src[i].seqno() != dst[i].seqno() ? kDiffSeq : 0;
-        // if cache seqno is different in src and destination
-        int cache_diff_flag =
```

Comment:
So this is no longer needed?

Reply:
No, we don't need this anymore, will remove. 
Because we now don't allow write to cache after loadcheckpoint called after we remove is_bootstrap parameter from last pr

- If worker exit before loadcheckpoint, it start from beginning and get full copy of cache when it calls allreduce/broadcast; 
- if worker exit after loadcheckpoint and before first allreduce, it should functions exactly same as previous case; 
- if worker exit after loadcheckpoint and after first allreduce/broadcat, allreduce consensus protocol grantee the diff_seq() will be set & since recover worker doesn't finish loadcheckpoint yet, we know this is cache recover; if loadcheckpoint called, we no longer call load_cache hence not hitting this code.

Reply:
Thanks for the detailed explanation.